polynote-spark/src/main/scala/polynote/kernel/LocalSparkKernel.scala

```diff
@@ -90,29 +93,43 @@ class LocalSparkKernelFactory extends Kernel.Factory.Service {
-    _                     <- Logging.info(s"Using spark classpath: ${sparkClasspath.mkString(":")}")
-    settings               = ScalaCompiler.defaultSettings(new Settings(), sparkRuntimeJar :: scalaDeps.map(_._2) ::: sparkClasspath)
-    sparkJars              = (sparkRuntimeJar :: ScalaCompiler.requiredPolynotePaths).map(f => f.toString -> f) ::: scalaDeps
-    sessionAndClassLoader <- startSparkSession(sparkJars, settings)
-    (session, classLoader) = sessionAndClassLoader
-    notebookPackage        = s"$$notebook$$${kernelCounter.getAndIncrement()}"
-    compiler              <- ScalaCompiler(settings, ZIO.succeed(classLoader), notebookPackage).map(ScalaCompiler.Provider.of)
-    busyState             <- SignallingRef[Task, KernelBusyState](KernelBusyState(busy = true, alive = true))
-    interpreters          <- RefMap.empty[String, Interpreter]
-    scalaInterpreter      <- interpreters.getOrCreate("scala")(ScalaSparkInterpreter().provide(compiler))
-    interpState           <- Ref[Task].of[State](State.predef(State.Root, State.Root))
+    scalaDeps        <- CoursierFetcher.fetch("scala")
+    sparkRuntimeJar   = new File(pathOf(classOf[SparkReprsOf[_]]).getPath)
+    sparkClasspath   <- (sparkClasspath orElse systemClasspath).option.map(_.getOrElse(Nil))
+    _                <- Logging.info(s"Using spark classpath: ${sparkClasspath.mkString(":")}")
+    settings          = ScalaCompiler.defaultSettings(new Settings(), sparkRuntimeJar :: scalaDeps.map(_._2) ::: sparkClasspath)
+    _                 = settings.outputDirs.setSingleOutput(new PlainDirectory(new Directory(org.apache.spark.repl.Main.outputDir)))
+    sparkJars         = (sparkRuntimeJar :: ScalaCompiler.requiredPolynotePaths).map(f => f.toString -> f) ::: scalaDeps
+    compiler         <- ScalaCompiler.provider(sparkRuntimeJar :: scalaDeps.map(_._2) ::: sparkClasspath)
+    classLoader      <- compiler.scalaCompiler.classLoader
+    session          <- startSparkSession(sparkJars, classLoader)
+    notebookPackage   = s"$$notebook$$${kernelCounter.getAndIncrement()}"
+    busyState        <- SignallingRef[Task, KernelBusyState](KernelBusyState(busy = true, alive = true))
+    interpreters     <- RefMap.empty[String, Interpreter]
+    scalaInterpreter <- interpreters.getOrCreate("scala")(ScalaSparkInterpreter().provide(compiler))
+    interpState      <- Ref[Task].of[State](State.predef(State.Root, State.Root))
   } yield new LocalSparkKernel(compiler, session, interpState, interpreters, busyState)
 
-  private def startSparkSession(deps: List[(String, File)], settings: Settings): TaskR[BaseEnv with GlobalEnv with CellEnv, (SparkSession, AbstractFileClassLoader)] = {
+  private def startSparkSession(deps: List[(String, File)], classLoader: ClassLoader): TaskR[BaseEnv with GlobalEnv with CellEnv, SparkSession] = {
+    def mkExecutor(): TaskR[Blocking, Executor] = effectBlocking {
```

Comment:
Was hoping to limit the proliferation of thread pools, but in this case it seems cleaner than setting some arbitrary thread's context classloader.